habitat/utils/gym_adapter.py

```diff
@@ -49,6 +51,96 @@ def transform_shape(shape):
+    """
+    if isinstance(original_space, spaces.Box):
+        # Any Box means it is continuous
+        return True
+    if isinstance(original_space, EmptySpace):
+        return False
+    if isinstance(original_space, Mapping):
+        return any([_is_continuous(v) for v in original_space.values()])
+    raise NotImplementedError(
+        f"Unknow action space found : {original_space}. Can only be Box or Empty"
+    )
+
+
+def _recursive_continuous_size_getter(
+    original_space: gym.Space, size: List[int]
+):
+    """
+    Returns the size of a continuous action vector from a habitat environment action space
+    """
+    if isinstance(original_space, spaces.Box):
+        size[0] += original_space.shape[0]
+    elif isinstance(original_space, EmptySpace):
+        size[0] += 1
+    elif isinstance(original_space, Mapping):
+        for v in original_space.values():
+            _recursive_continuous_size_getter(v, size)
+    else:
+        raise NotImplementedError(
+            f"Unknow continuous action space found : {original_space}. Can only be Box, Empty or Dict."
+        )
+
+
+def create_action_space(original_space: gym.Space) -> gym.Space:
+    """
+    Converts a habitat task action space into a either continuous (Box) or discrete (Discrete) gym.space.
+    """
+    assert isinstance(
+        original_space, Mapping
+    ), f"The action space of the environment needs to be a Mapping, but was {original_space}"
+    if _is_continuous(original_space):
+        size = [0]
+        _recursive_continuous_size_getter(original_space, size)
+        return spaces.Box(
+            shape=(size[0],), low=-1.0, high=1.0, dtype=np.float32
+        )
+    else:
+        # discrete case. The ActionSpace class gives us the correct action size
+        return spaces.Discrete(len(original_space))
+
+
+def continuous_vector_action_to_hab_dict(
+    action_space, action: np.ndarray, clip: bool = True
+) -> Dict[str, Any]:
+    """We naively assume that all actions are 1D (len(shape) == 1)"""
+
+    # Assume that the action space only has one root SimulatorTaskAction
+    root_action_names = tuple(action_space.spaces.keys())
+    if len(root_action_names) == 1:
+        # No need for a tuple if there is only one action
+        root_action_names = root_action_names[0]
+    action_name_to_lengths = {}
+    for outer_k, act_dict in action_space.spaces.items():
+        if isinstance(act_dict, EmptySpace):
+            action_name_to_lengths[outer_k] = 1
+        else:
+            for k, v in act_dict.items():
+                # The only element in the action
+                action_name_to_lengths[k] = v.shape[0]
+
+    # Determine action arguments for root_action_name
+    action_args = {}
+    action_offset = 0
+    for action_name, action_length in action_name_to_lengths.items():
+        action_values = action[action_offset : action_offset + action_length]
+        if clip:
+            action_values = np.clip(action_values, -1.0, 1.0)
```

Comment:
I am against having this clip. I thought about it more, and the clip should be inside the environment, not hidden in the action conversion. Not all environments expect clipped actions I think. All the habitat 2.0 continuous action spaces already do clipping internally. I think this should be removed.